WireAPI/Sources/WireAPI/APIs/ConversationsAPI/ConversationsAPI.swift

```diff
@@ -0,0 +1,29 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+/// Access to conversations API.
+public protocol ConversationsAPI {
+
+    /// Fetch all conversation identifiers in batches for ``APIVersion`` v0.
+    func getLegacyConversationIdentifiers() async throws -> PayloadPager<UUID>
+
+    /// Fetch all conversation identifiers in batches available from ``APIVersion`` v1.
+    func getConversationIdentifiers() async throws -> PayloadPager<QualifiedID>
```

Comment:
I think we need a comment here to remember why we ended up with a two methods for same need.

Normally we would have one method that abstract the api version away.

Reply:
Okay, I can add a comment in the implementation that mapping was a problem, but I think here for the caller it's clear how to use the API object, isn't it?

Reply:
Normally yes, but we have different return types that can not be simply mapped to each other... and passing the backend version was an egg dance... as you can see in the last commits. So I would prefer to keep it up to the caller now how to process the results, ok?

Reply:
yes make sense, my point was more about the why, adding a comment in the implementation sounds good :)

Reply:
0c74973272eb45186bfbdf666a262354cdebfecd